### DISCUSSION TITLE
which_key in visual mode?

### DISCUSSION THREAD

**Discussion by poncito** [LunarVim/LunarVim#4422] [Q&A] [answered]
Hi, in my nvim config, I set some keybindings with whichkey, in visual mode:
```lua
which_key.register({
    r = {
        name="Slime",
        r = {"<Plug>SlimeRegionSend", "Send region"},
        c = { "<Plug>SlimeConfig", "Config" },
    }
}, { prefix="<leader>", mode="v"})
```
is there a way to do the same with lvim?
Thanks!

**Comment by DevNightlion** [comment, marked as answer]
Hi poncito,
To add or remap keybindings for whichkey use `lvim.builtin.which_key.mappings`. See official lvim documentation [lvim - whichkey-bindings](https://www.lunarvim.org/docs/configuration/keybindings#whichkey-bindings).

Unfortunately the documentation is missing information about `vmappings` which is for setting which-key in visual mode (see [which-key.lua#L98](https://github.com/LunarVim/LunarVim/blob/9dc549a4edd4b79f7f3bcad0e386e490a6a2b952/lua/lvim/core/which-key.lua#L98)

The configuration from your case could look like this
```lua
lvim.builtin.which_key.vmappings["r"] = {
  name = "Slime",
  r = {"<Plug>SlimeRegionSend", "Send region"},
  c = { "<Plug>SlimeConfig", "Config" },
}
```

**Reply by poncito** [reply]
Thank you!
